### ISSUE TITLE
Why separate edge case for Migration token delegate role?

### ISSUE THREAD

**Issue by fedorerd** [metaplex-foundation/metaplex-program-library#1091]
I just faced some issues working with `revoke` within my own program linked to recently migrated NFTs. I assume similar problem exists with other instructions also. 

So, I have a staking program and I am invoking `revoke` with `RevokeArgs::StakingV1`. Since PNFTs are locked with Staking delegate role. However, if I pass `RevokeArgs::StakingV1` for recently migrated NFT (which has `Migration` delegate role), instructions will fail here:
https://github.com/metaplex-foundation/metaplex-program-library/blob/3a9e2eac5219eb20eccdd7a762fe24e6867c28f3/token-metadata/program/src/processor/delegate/revoke.rs#L243

Your program checks if role provided in revoke args matches actual delegate role or not. However, I don't understand why is it so, since Migration delegate has same rules for revoke as other delegates. 

I'm talking only about migration delegate, since it is one-time state for any token. 

So my program usually will pass `StakingV1` arg and expect staking delegate role, however due to design I need to add an edge case in my program and check status of delegate in token record on my side. I think it would be much better for everyone if this edge case was checked on your side instead.

And there is no evidence for how long I will need to keep this edge case check, since I literally need all migrated while staked in my program NFTs to be unstaked once.

So, basically, I think check like that might be more convenient:
```rust
if token_record.delegate_role == Some(role) || token_record.delegate_role == Some(TokenDelegateRole::Migration) {
  ...
}
```

**Comment by samuelvanderwaal** [comment]
@fedorerd This is a good point and I opened a PR to fix it.

**Comment by samuelvanderwaal** [comment]
This fix is on mainnet now in the 1.11.2 release.
